waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -1340,26 +1387,25 @@ proc mount(node: WakuNode,
       # do not persist or use a persisted rln-relay credential
       # a new credential will be generated during the mount process but will not be persisted
       info "no need to persist or use a persisted rln-relay credential"
-      res = waitFor node.mountRlnRelayDynamic(memContractAddr = ethMemContractAddress, ethClientAddr = ethClientAddr,
+      res = await node.mountRlnRelayDynamic(memContractAddr = ethMemContractAddress, ethClientAddr = ethClientAddr,
                 ethAccountAddress = ethAccountAddressOpt, ethAccountPrivKeyOpt = ethAccountPrivKeyOpt, pubsubTopic = conf.rlnRelayPubsubTopic,
                 contentTopic = conf.rlnRelayContentTopic, spamHandler = spamHandler, registrationHandler = registrationHandler)
       
     if res.isErr():
-        return err("dynamic rln-relay could not be mounted: " & res.error())
-    return ok(res.value())
+      return err("dynamic rln-relay could not be mounted: " & res.error())
+    return ok()
 
 
 proc mountRlnRelay*(node: WakuNode,
                     conf: WakuNodeConf|Chat2Conf,
                     spamHandler: Option[SpamHandler] = none(SpamHandler),
                     registrationHandler: Option[RegistrationHandler] = none(RegistrationHandler)
-                   ): RlnRelayResult[bool] {.raises: [Defect, ValueError, IOError, CatchableError, Exception].} =
-  waku_rln_relay_mounting_duration_seconds.nanosecondTime:
-    let res = mount(
+                   ): Future[RlnRelayResult[void]] {.async.} =
```

Comment:
why do we need this change?

Reply:
1. We don't really need to use bool for a result unless necessary. In this case, we just need to check if it mounted successfully, which is indicated by a success result, i.e ok()
2. It is a Future now since we moved from using waitFor to async processing